lost/src/test/java/com/mapzen/android/lost/internal/LocationSettingsResultRequestTest.java

```diff
@@ -0,0 +1,606 @@
+    when(pm.hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_NETWORK)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_GPS)).thenReturn(true);
+
+    ArrayList<LocationRequest> requests = new ArrayList<>();
+    LocationRequest noPower = LocationRequest.create().setPriority(
+        LocationRequest.PRIORITY_NO_POWER);
+    requests.add(noPower);
+    LocationSettingsRequest settingsRequest = new LocationSettingsRequest.Builder()
+        .addAllLocationRequests(requests)
+        .setNeedBle(true)
+        .build();
+    LocationSettingsResultRequest settingsResultRequest = new LocationSettingsResultRequest(
+        bluetoothAdapter, pm, locationManager, generator, settingsRequest);
+
+    LocationSettingsResult result = settingsResultRequest.await();
+    assertThat(result.getStatus().getStatusCode()).isEqualTo(Status.RESOLUTION_REQUIRED);
+  }
+
+  @Test
+  public void awaitTimeout_shouldTimeoutAfterInterval() {
+    when(pm.hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_NETWORK)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_GPS)).thenReturn(true);
+
+    DelayTestPendingIntentGenerator delayedGenerator = new DelayTestPendingIntentGenerator(context);
+    LocationSettingsResultRequest settingsResultRequest = new LocationSettingsResultRequest(
+        bluetoothAdapter, pm, locationManager, delayedGenerator, request);
+
+    LocationSettingsResult resultRequest = settingsResultRequest.await(1000, TimeUnit.MILLISECONDS);
+    assertThat(resultRequest.getStatus().getStatusCode()).isEqualTo(Status.TIMEOUT);
+  }
+
+  @Test
+  public void cancel_shouldReturnCancelResult() {
+    when(pm.hasSystemFeature(PackageManager.FEATURE_BLUETOOTH_LE)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_NETWORK)).thenReturn(true);
+    when(pm.hasSystemFeature(PackageManager.FEATURE_LOCATION_GPS)).thenReturn(true);
+    when(locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)).thenReturn(true);
+    when(locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)).thenReturn(true);
+    when(bluetoothAdapter.isEnabled()).thenReturn(true);
+
+    DelayTestPendingIntentGenerator delayedGenerator = new DelayTestPendingIntentGenerator(
+     context);
+    final LocationSettingsResultRequest settingsResultRequest = new LocationSettingsResultRequest(
+     bluetoothAdapter, pm, locationManager, delayedGenerator, request);
+
+    Thread thread = new Thread(new Runnable() {
+      @Override public void run() {
+        settingsResultRequest.setResultCallback(new ResultCallback<LocationSettingsResult>() {
+          @Override public void onResult(@NonNull LocationSettingsResult result) {
+            assertThat(result.getStatus().getStatusCode()).isEqualTo(Status.CANCELLED);
```

Comment:
I don't have a good way to cleanly test this at the moment so I've removed the test. Let me know if you see a better way forward though.